### PR TITLE
Add int64 input type for ReduceProd

### PR DESCRIPTION
**Pull request by neginraoof**
**Description**:
Adding int64 input type for ReduceProd

**Motivation and Context**
This input type is required for exporting numel op.
Numel op applies ReduceProd on the output of the Shape node. Output of the Shape node is always of type int64, which is expected since it represents sizes.

Numel is a frequently used operator that's used in torchvision models Mask RCNN and Keypoint RCNN.

### CODE REVIEW

**Comment by neginraoof** [comment]
@hari90 Would you please help review this PR?
Thanks

File: onnxruntime/core/providers/cuda/reduction/reduction_ops.cc [hunk 1]
```diff
@@ -887,5 +887,6 @@ REGISTER_KERNEL_INT32(ReduceMean)
 REGISTER_KERNEL_INT32(ReduceProd)
 REGISTER_KERNEL_INT32(ReduceSum)
 
+REGISTER_KERNEL_TYPED_12(ReduceProd, int64_t)
```
Comment (hariharans29) [inline onnxruntime/core/providers/cuda/reduction/reduction_ops.cc:890]: Wouldn't this try to register for opset-12 as well ? (Not sure if this is causing the build failures as I don't think ORT has taken in the ONNX commits pertaining to opset-12 yet)

Comment (neginraoof) [reply onnxruntime/core/providers/cuda/reduction/reduction_ops.cc]: Oh I see. Fixed it. Thanks a lot.

**Comment by hariharans29** [comment]
Regarding the CUDA build breaks - can you pls check if the cudnn library function being used in the cuda implementation supports int64 ? If it doesn't, you can just skip adding a kernel for CUDA. It should fallback to CPU for the data type.

File: onnxruntime/core/providers/cuda/reduction/reduction_functions.cu [hunk 2]
```diff
@@ -396,6 +396,8 @@ template void reduce_matrix_rows<float, float>(
   const float* data, float* output, int m, int n);
 template void reduce_matrix_rows<double, double>(
   const double* data, double* output, int m, int n);
+template void reduce_matrix_rows<int64_t, int64_t>(
+  const double* data, double* output, int m, int n);
```
Comment (hariharans29) [inline onnxruntime/core/providers/cuda/reduction/reduction_functions.cu:400]: double or int64 ?

Comment (hariharans29) [reply onnxruntime/core/providers/cuda/reduction/reduction_functions.cu]: Probably both data and output are int64 (going based on the pattern above)

Comment (neginraoof) [reply onnxruntime/core/providers/cuda/reduction/reduction_functions.cu]: Yes, got it. Looks after fixing this, the exception was:
 Error : no instance of function template "onnxruntime::cuda::reduce_matrix_rows" matches the specified type

I have changed it to CPU only for now.

**Comment by neginraoof** [comment]
@snnn I'm unable to see the review from you. Would you please let me know if I'm missing anything?

**Pull request merged**